### PR TITLE
docs/user: add introductory guide for spill2disk

### PR DESCRIPTION
**Pull request by guswynn**
This is broken out of #21209, removing most (but not all) of the sizing guidance in the original pr (this is now in https://github.com/MaterializeInc/materialize/pull/21491). It still attempts to be fairly thorough

### Motivation

docs

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - add guide for using spill2disk (private-preview)

### CODE REVIEW

**Comment by sjwiesman** [comment]
@guswynn I've pushed an update to your PR. Please take a look.

File: doc/user/content/ingest-data/spill-to-disk.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+{{< /warning >}}
+
+This is a guide on how and why to run Upsert and Debezium Kafka sources with spill to disk.
+
+## Overview
+
+Kafka sources that use `ENVELOPE UPSERT` or `ENVELOPE DEBEZIUM` require the entire _working set_ to be stored locally within its scheduled cluster to produce retractions when keys are updated.
+The working set is sum total of _current_ values for every key.
+
+By default, Materialize uses an in-memory map to store the working set. However, sources scheduled to run on a cluster with an attached disk store this state on that attached disk. Disk-enabled clusters can reduce source memory usage, allowing users to leverage smaller cluster sizes.
```
Comment (guswynn) [inline doc/user/content/ingest-data/spill-to-disk.md:24]: One nit:

while we want to use `spill-to-disk` for the feature, when we actually are talking about replicas, we use "disk-attached replicas", and we should link to this section: https://materialize.com/docs/sql/create-cluster-replica/#disk-attached-replicas here!

**Review by guswynn — COMMENTED** [review]
I can't accept my own pr, but changes look good

**Comment by morsapaes** [comment]
Sorry for the delay, @guswynn! I'll push some suggestions to blend in the spill-to-disk guidance in the guide into existing guides, so it's easier to use in the right context. I can take it from here — thanks for all the work you've put into documenting this new feature!

**Comment by guswynn** [comment]
thanks @morsapaes !

**Comment by guswynn** [comment]
@morsapaes do you mind taking a look at this again?

**Comment by morsapaes** [comment]
@sjwiesman @benesch: could you give this another pass? Tried to condense the original guide into a succinct section in [`kafka.md`](https://github.com/MaterializeInc/materialize/pull/21438/commits/da407a22feb27224ff8021a299608011f8589d79#diff-3b8796ce89724686a9aba8486483aa6be62e83c91586458b47f1aff075f9b603) and link out to the cluster reference documentation for more details.

**Pull request merged**